lumicks/pylake/kymotracker/kymotracker.py

```diff
@@ -60,6 +64,7 @@ def track_greedy(
     sigma_cutoff=2.0,
     rect=None,
     line_width=None,
+    get_arg_defaults=False,
```

Comment:
While this will work, this seems pretty smelly to be honest. I don't like how the function gets two distinct tasks now.

To me, it would make sense if whatever provided the defaults was either a separate function or an annotation (for simple cases). We could have some testing in place that verifies that for functions where we define such things, the defaults are compatible with the defaults in the actual function signature (where applicable).

For simple function arguments (fixed ones) starting from Python `3.9` we could do something like:

```python
from dataclasses import dataclass
from typing import Annotated, Union
​
@dataclass
class Parameter:
    min: float
    max: float
    description: str
​
def function(
    number: Union[float, int],
    fraction: Annotated[float, Parameter(0.0, 1.0, "cool")] = 0.5
):
    pass
​
function(3, 4)
function.__annotations__
{'number': typing.Union[float, int],
 'fraction': typing.Annotated[float, Parameter(min=0.0, max=1.0, description='cool')]}
```
LV could just read those from the `__annotation__` dunder then. For non-simple ones, I'm not sure whether we've seen enough of them to warrant a general solution yet. I'd likely prefer to have the general "good guesstimator" in a separate function, and have us call it in the actual function.